### PR TITLE
Update Manifest descriptions

### PR DESCRIPTION
**Pull request by agriffard**
Provide a more detailled description for some modules.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.DynamicCache/Manifest.cs [hunk 1]
```diff
@@ -5,6 +5,6 @@
     Author = ManifestConstants.OrchardCoreTeam,
     Website = ManifestConstants.OrchardCoreWebsite,
     Version = ManifestConstants.OrchardCoreVersion,
-    Description = "Dynamic Cache.",
+    Description = "The Dynamic Cache module allows to use tag helpers to cache items dynamically.",
```
Comment (sfmskywalker) [inline src/OrchardCore.Modules/OrchardCore.DynamicCache/Manifest.cs:8]: Or perhaps: `"The Dynamic Cache module adds tag helpers to cache items dynamically."`?

File: src/OrchardCore.Modules/OrchardCore.Navigation/Manifest.cs [hunk 2]
```diff
@@ -5,7 +5,7 @@
     Author = ManifestConstants.OrchardCoreTeam,
     Website = ManifestConstants.OrchardCoreWebsite,
     Version = ManifestConstants.OrchardCoreVersion,
-    Description = "Navigation.",
+    Description = "The Navigation module allows to define menus and display them.",
```
Comment (sfmskywalker) [inline src/OrchardCore.Modules/OrchardCore.Navigation/Manifest.cs:8]: Or perhaps: "The Navigation module allows **you** to define menus and display them."?

File: src/OrchardCore.Modules/OrchardCore.ResponseCompression/Manifest.cs [hunk 3]
```diff
@@ -5,6 +5,6 @@
     Author = ManifestConstants.OrchardCoreTeam,
     Website = ManifestConstants.OrchardCoreWebsite,
     Version = ManifestConstants.OrchardCoreVersion,
-    Description = "Compresses responses.",
+    Description = "The Response Compression module allows to enable the gzip compression.",
```
Comment (sfmskywalker) [inline src/OrchardCore.Modules/OrchardCore.ResponseCompression/Manifest.cs:8]: "The Response Compression module allows you to enable gzip compression"

File: src/OrchardCore.Modules/OrchardCore.Navigation/Manifest.cs (hunk 2 above)
Comment (agriffard) [reply src/OrchardCore.Modules/OrchardCore.Navigation/Manifest.cs:8]: ok

File: src/OrchardCore.Modules/OrchardCore.DynamicCache/Manifest.cs (hunk 1 above)
Comment (agriffard) [reply src/OrchardCore.Modules/OrchardCore.DynamicCache/Manifest.cs:8]: ok

**Pull request merged**